### DISCUSSION TITLE
Example on training with TPU does not run at all

### DISCUSSION THREAD

**Discussion by tungts1101** [Lightning-AI/pytorch-lightning#9966] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am currently try this colab notebook https://colab.research.google.com/github/PytorchLightning/lightning-tutorials/blob/publication/.notebooks/lightning_examples/mnist-tpu-training.ipynb#scrollTo=2772a2e1 provided by PL teams to get some experience with TPU training. But when I try to execute the third cell, there is some Import error with the _XLAC module. 

<img width="937" alt="Screen Shot 2021-10-17 at 01 44 00" src="https://user-images.githubusercontent.com/37470762/137598792-4478d28d-485a-45da-9186-947387a5394a.png">

**Comment by prdai** [comment]
hi,
the problem seems to be with torch_xla,

try

`!pip install cloud-tpu-client==0.10 https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.9-cp37-cp37m-linux_x86_64.whl`

`!pip install pytorch-lightning`

if you have cuda 10.2
`!pip3 install torch torchvision torchaudio`

if you have cuda 11.1
`!pip3 install torch==1.9.1+cu111 torchvision==0.10.1+cu111 torchaudio==0.9.1 -f https://download.pytorch.org/whl/torch_stable.html`

and you can add an issue to `https://github.com/pytorch/xla`

which is the library which seems to have the problem can you try these.

With best regards,
Ranuga

**Reply by prdai** [reply]
> hi, the problem seems to be with torch_xla,
> 
> try
> 
> `!pip install cloud-tpu-client==0.10 https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.9-cp37-cp37m-linux_x86_64.whl`
> 
> `!pip install pytorch-lightning`
> 
> if you have cuda 10.2 `!pip3 install torch torchvision torchaudio`
> 
> if you have cuda 11.1 `!pip3 install torch==1.9.1+cu111 torchvision==0.10.1+cu111 torchaudio==0.9.1 -f https://download.pytorch.org/whl/torch_stable.html`
> 
> and you can add an issue to `https://github.com/pytorch/xla`
> 
> which is the library which seems to have the problem can you try these.
> 
> With best regards, Ranuga

I will check the notebook and try to resolve it

With best regards,
Ranuga

**Reply by prdai** [reply]
> `!pip3 install torch==1.9.1+cu111 torchvision==0.10.1+cu111 torchaudio==0.9.1 -f https://download.pytorch.org/whl/torch_stable.html`

Check this issue also
https://stackoverflow.com/questions/68846290/colab-pytorch-importerror-usr-local-lib-python3-7-dist-packages-xlac-cpytho

**Reply by prdai** [reply]
hi,

I found another issue is similar

check the issue below
https://github.com/pytorch/xla/issues/2259

it seems to have the fix for your issue

With best regards,
Ranuga

**Reply by prdai** [reply]
check this issue also

https://github.com/facebookresearch/maskrcnn-benchmark/issues/891

**Comment by akihironitta** [comment]
@tungts1101 Hi, please have a look at #8315.

**Comment by kaushikb11** [comment, marked as answer]
Hi @tungts1101! this error is raised when the PyTorch and PyTorch xla are not of the same versions.
You could verify using `pip list | grep torch` and could install the latest versions for both!
